core/sqf/src/seatrans/tm/hbasetmlib2/src/main/java/org/trafodion/dtm/HBaseTxClient.java

```diff
@@ -1062,6 +1062,14 @@ public void run() {
                                               tmID + " regionBytes: [" + regionBytes + "].  Deleting zookeeper region entry. \n exception: " + de);
                                       zookeeper.deleteRegionEntry(regionEntry);
                                 }// DeserializationException
+                                catch (IOException ioe) {
+                                    // It's possible we received a spurious entry from a stale region.
+                                       // It's safe to delete the zookeeper entry and wait for any new posts
+                                       LOG.warn("TRAF RCOV THREAD:IOException calling txnManager.recoveryRequest. " + "TM: " +
+                                               tmID + " regionBytes: [" + regionBytes
+                                               + "].  Deleting zookeeper region entry. exception: " + ioe);
+                                       zookeeper.deleteRegionEntry(regionEntry);
+                                 }// IOException
```

Comment:
TableNotFoundException is a non-issue, however, in the case of NotServingRegionException, you are correct.  I will look at addressing this.